addons/controllers/webhooks_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	cert2 "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"knative.dev/pkg/webhook/certificates/resources"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/controllers/webhooks"
+)
+
+func TestWebhooks(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Webhooks TLS",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var (
+	setupLog = ctrl.Log.WithName("controllers").WithName("Addon")
+)
+var _ = Describe("when TLS secret is created", func() {
+
+	Context("If the certificates do not exists", func() {
```

Comment:
nit: usually lower case. e.g., `if xxxx`

Reply:
not nit. You are correct. it should be lower

Reply:
done